packages/template-retail-react-app/app/pages/account/orders.test.js

```diff
@@ -93,3 +93,146 @@ test('Renders order history place holder when no orders', async () => {
+        expect(await screen.findByTestId('account-order-details-page')).toBeInTheDocument()
+    })
+
+    test('should show 0 items', async () => {
+        await user.click((await screen.findAllByText(/view details/i))[0])
+        expect(await screen.findByText(/0 items/i)).toBeInTheDocument()
+    })
+
+    test('should not render products', async () => {
+        await user.click((await screen.findAllByText(/view details/i))[0])
+        expect(screen.queryByAltText(/Pleated Bib Long Sleeve Shirt/i)).not.toBeInTheDocument()
+    })
+})
+
+describe('Direct navigation to order details and back to order list', () => {
+    let user, orderNo
+    beforeEach(async () => {
+        global.server.use(
+            rest.get('*/orders/:orderNo', (req, res, ctx) => {
+                return res(ctx.delay(0), ctx.json(mockOrderHistory.data[0]))
+            }),
+            rest.get('*/customers/:customerId/orders', (req, res, ctx) => {
+                return res(ctx.delay(0), ctx.json(mockOrderHistory))
+            }),
+            rest.get('*/products', (req, res, ctx) => {
+                return res(ctx.delay(0), ctx.json(mockOrderProducts))
+            })
+        )
+        orderNo = mockOrderHistory.data[0].orderNo
+        window.history.pushState(
+            {},
+            'Order Details',
+            createPathWithDefaults(`/account/orders/${orderNo}`)
+        )
+        const renderResult = renderWithProviders(<MockedComponent history={history} />, {
+            wrapperProps: {siteAlias: 'uk', appConfig: mockConfig.app}
+        })
+        user = renderResult.user
+    })
+
+    test('should render order details page on direct navigation', async () => {
+        expect(await screen.findByTestId('account-order-details-page')).toBeInTheDocument()
+        expect(window.location.pathname).toMatch(new RegExp(`/account/orders/${orderNo}$`))
+    })
+
+    test('should navigate back to order history page', async () => {
+        await user.click(await screen.findByRole('link', {name: /back to order history/i}))
+        expect(await screen.findByTestId('account-order-history-page')).toBeInTheDocument()
+        expect(window.location.pathname).toMatch(/\/account\/orders$/)
+    })
+
+    test('should show all orders', async () => {
+        await user.click(await screen.findByRole('link', {name: /back to order history/i}))
+        expect(await screen.findAllByText(/Ordered: /i)).toHaveLength(3)
+    })
+
+    test('should show all products', async () => {
+        await user.click(await screen.findByRole('link', {name: /back to order history/i}))
+        expect(
+            await screen.findAllByAltText(
+                'Pleated Bib Long Sleeve Shirt, Silver Grey, small',
+                {},
+                {timeout: 500}
+            )
+        ).toHaveLength(3)
+    })
+})
+
+describe('Handles order with missing or partial data gracefully', () => {
+    let orderNo
+    beforeEach(async () => {
+        const partialOrder = {
```

Comment:
I am wondering if this is a real case scenario. From the storefront, if you don't input your billing shipment or payment, you will never be able to check out. I think the API will not allow you to check out without these missing information either.

Reply:
Understood but the get api could have a bug in it.